services/proxy/pkg/userroles/oidcroles.go

```diff
@@ -29,6 +31,58 @@ func NewOIDCRoleAssigner(opts ...Option) UserRoleAssigner {
 	}
 }
 
+func extractRoles(rolesClaim string, claims map[string]interface{}) (map[string]struct{}, error) {
+
+	claimRoles := map[string]struct{}{}
+	// happy path
+	value, _ := claims[rolesClaim].(string)
+	if value != "" {
+		claimRoles[value] = struct{}{}
+		return claimRoles, nil
+	}
+
+	// try splitting path at .
+	segments := oidc.SplitWithEscaping(rolesClaim, ".", "\\")
+
+	subclaims := claims
+	lastSegment := len(segments) - 1
+	for i := range segments {
+		if i < lastSegment {
```

Comment:
Same question as above. Can't we avoid the `if ... else` in the already complex for loop? Or better even call code from `oidc` package?